packages/wrangler/src/queues/cli/commands/consumer/add.ts

```diff
@@ -37,6 +37,15 @@ export function options(yargs: CommonYargsArgv) {
 				type: "string",
 				describe: "Queue to send messages that failed to be consumed",
 			},
+			"concurrency-enabled": {
```

Comment:
Why is this called `concurrency-enabled`? If it's a boolean, could it just be `--concurrency`?

Although actually, what about a single flag `--concurrency` an optional numeric argument?

Reply:
I suppose we could call it `concurrency` /shrug.

There are currently two options to support the following:
* The user can decide whether they want concurrency turned on or off entirely
* If concurrency is turned on, the user can *optionally* specify a maximum concurrency level (or omit it / pass null to indicate that they do not care to enforce a maximum level).

Reply:
To add to what Josh described: our goal is to have `concurrency_enabled` be the default, and to (ideally) have only users that need to strictly limit set `false` or `max_concurrency: $SOME_FIXED_NUMBER`.

Over time, as we allow more concurrent invocations, developers will just benefit from those improvements without having to touch their config.

Reply:
If you merge the two properties, how do you express "enable concurrency, and let Cloudflare define the maximum" (which will grow over time) - e.g. effectively an `auto` mode. 

In this proposal, that's just `concurrency_enabled: true` (or `null` / omit it). `max_concurrency` is a control for users who have upstream concerns.

"Technically" this could be a union type but that's not easily represented in TOML nor is it familiar to most users.

Reply:
After some internal discussion, we went ahead and consolidated down to a single `max_concurrency` setting.
Behind the scenes, a missing/null `max_concurrency` will fallback to a default value of 1, temporarily, for the next few weeks.
After communicating the change, the fallback/default will change to 'unlimited' mode.